src/wxProfilerGUI/database.cpp

```diff
@@ -243,6 +243,45 @@ void Database::loadSymbols(wxInputStream &file)
 	progressdlg.Update(kMaxProgress, "Tidying things up...");
 }
 
+static void addSampleInfo(std::vector<Database::SampleInfo> &samples, Database::SampleInfo const &info)
+{
+	auto it = std::lower_bound(samples.begin(), samples.end(), info, [](Database::SampleInfo const &i0, Database::SampleInfo const &i1) { return i0.threadid < i1.threadid; });
+	if (it == samples.end() || it->threadid != info.threadid) {
+		samples.insert(it, info);

```

Comment:
Inserting into a vector is an O(n) operation. I'm not sure how many of those we've had before, but if this is new, it may result in a performance regression for longer profiling sessions.

Reply:
New entries are only inserted if a sample is taken from the context of a new thread. If the profiled application starts and retires an unbounded number of threads, and we encounter callstack samples called from those threads, then yes - inserting might become an issue. Currently the profiled thread number is constant however. A potential solution to improve the O() complexity would be to use a map instead of a vector. Given the implementation of a map as a tree is not cache-friendly, I chose to use an array since that would be faster for the current/typical use case. If you're more concerned with the worst, rather than the typical case performance however, I can change the array to a map?

Reply:
Thanks for the explanation. You're probably right, we (or the user otherwise) will probably start running into performance problems elsewhere before this becomes an issue.

Reply:
Changed the storage to a map instead of sorted array, it should fix the potential worst case behavior while the typical case slowdown should not be noticeable to the user.